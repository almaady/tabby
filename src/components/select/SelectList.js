import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import SelectListItem from "./SelectListItem";

/**
 * @class SelectList - The list of options of the select
 */
class SelectList extends React.PureComponent {
  /**
   * @property {string} className - ClassName if you need to customize
   * @property {func} onClick - OnClick function for each option
   * @property {array} options - List of options
   */
  static propTypes = {
    className: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    // eslint-disable-next-line react/forbid-prop-types
    options: PropTypes.array.isRequired
  };

  render() {
    const { className, options, onClick, selectedValue} = this.props;

    return ( 
      <ul className={classNames(className, "select__list")}>
        {options.map((item, index) => (
          <SelectListItem 
            icon={item.icon} 
            key={index} 
            title={item.title} 
            selected={item.selected} 
            onClick={onClick} 
            value={item.value}
            selectedValue={selectedValue}
            />
        )) } 
      </ul>
    );
  }
}

SelectList.defaultProps = {
  icon: false,
  className: ""
};

export default SelectList