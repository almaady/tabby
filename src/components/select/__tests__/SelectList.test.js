import React from 'react';
import {shallow} from "enzyme";
import sinon from "sinon"

import SelectList from "../SelectList";

describe(SelectList, () => {
  const click = sinon.stub();
  let props;
  beforeEach(() => {
    props = {
      onClick: click,
      options: [
        {
          title: "Yo soy la opción 1",
          icon: "top",
          id: "1",
          value: "option 1"
        },
        {
          title: "Yo soy la opción 2",
          icon: "left",
          id: "2",
          value: "option 2"
        },
        {
          title: "Yo soy la opción 3",
          icon: "right",
          id: "3",
          value: "option 3"
        }
      ]
    };
  });

  it("should render correctly with default state", () => {
    const component = shallow(<SelectList {...props} />);
    expect(component).toMatchSnapshot();
  });
  it("should has three node children", () => {
    const component = shallow(<SelectList {...props} />);
    expect(component.find('SelectListItem').length).toEqual(props.options.length);
  });
});