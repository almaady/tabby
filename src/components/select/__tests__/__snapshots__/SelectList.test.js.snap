// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectList should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectList
    className=""
    icon={false}
    onClick={[Function]}
    options={
      Array [
        Object {
          "icon": "top",
          "id": "1",
          "title": "Yo soy la opción 1",
          "value": "option 1",
        },
        Object {
          "icon": "left",
          "id": "2",
          "title": "Yo soy la opción 2",
          "value": "option 2",
        },
        Object {
          "icon": "right",
          "id": "3",
          "title": "Yo soy la opción 3",
          "value": "option 3",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SelectListItem
          className=""
          icon="top"
          onClick={[Function]}
          selectedValue=""
          title="Yo soy la opción 1"
          value="option 1"
        />,
        <SelectListItem
          className=""
          icon="left"
          onClick={[Function]}
          selectedValue=""
          title="Yo soy la opción 2"
          value="option 2"
        />,
        <SelectListItem
          className=""
          icon="right"
          onClick={[Function]}
          selectedValue=""
          title="Yo soy la opción 3"
          value="option 3"
        />,
      ],
      "className": "select__list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "className": "",
          "icon": "top",
          "onClick": [Function],
          "selected": undefined,
          "selectedValue": "",
          "title": "Yo soy la opción 1",
          "value": "option 1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "className": "",
          "icon": "left",
          "onClick": [Function],
          "selected": undefined,
          "selectedValue": "",
          "title": "Yo soy la opción 2",
          "value": "option 2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "className": "",
          "icon": "right",
          "onClick": [Function],
          "selected": undefined,
          "selectedValue": "",
          "title": "Yo soy la opción 3",
          "value": "option 3",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SelectListItem
            className=""
            icon="top"
            onClick={[Function]}
            selectedValue=""
            title="Yo soy la opción 1"
            value="option 1"
          />,
          <SelectListItem
            className=""
            icon="left"
            onClick={[Function]}
            selectedValue=""
            title="Yo soy la opción 2"
            value="option 2"
          />,
          <SelectListItem
            className=""
            icon="right"
            onClick={[Function]}
            selectedValue=""
            title="Yo soy la opción 3"
            value="option 3"
          />,
        ],
        "className": "select__list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "className": "",
            "icon": "top",
            "onClick": [Function],
            "selected": undefined,
            "selectedValue": "",
            "title": "Yo soy la opción 1",
            "value": "option 1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "className": "",
            "icon": "left",
            "onClick": [Function],
            "selected": undefined,
            "selectedValue": "",
            "title": "Yo soy la opción 2",
            "value": "option 2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "className": "",
            "icon": "right",
            "onClick": [Function],
            "selected": undefined,
            "selectedValue": "",
            "title": "Yo soy la opción 3",
            "value": "option 3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
